workspaces/ui-v2/src/optic-components/pages/diffs/ReviewDiffPages.tsx

```diff
@@ -46,6 +46,9 @@ export function DiffReviewPages(props: any) {
     allPaths.loading ||
     allRequestsAndResponsesOfBaseSpec.loading;
 
+  // TODO: we have a problem here because isLoading is false on mount,
+  // then true on next render and after that it becomes false upon loadin
+  // this results in a page content flash before it starts loading data
```

Comment:
I'm not sure I follow this - are you talking about initial mount, or a capture change? [On initial mount](https://github.com/opticdev/optic/blob/develop/workspaces/ui-v2/src/optic-components/hooks/useDiffForCapture.ts#L30) `diff.loading` is `true`.

Reply:
I was wrong about this... my bad. You are right, `isLoading` is true on mount.
My assumption was that `isLoading` was responsible for page content flash before displaying the loading indicator. To reproduce this go to Docs page, and then go to Diffs page. But this minor problem is caused by rendering `'/apis/:specId/diffs'` and then redirecting to `'/apis/:specId/diffs/local/:captureId'`. I added another commit to fix this

Reply:
Thanks for digging into this!